src/main/java/sk/freemap/gpxAnimator/TileCache.java

```diff
@@ -52,7 +52,7 @@ public static BufferedImage getTile( final String url, final String tileCachePat
 
         BufferedImage image;
         
-        if (tileCachePath == null) {
+        if ((tileCachePath == null) || (tileCachePath.length() < 1)) {
```

Comment:
To be *really* sure you could do `tileCachePath.trim().length() < 1`. Just in case whitespace found it's way into that field.